src/backend/postmaster/postmaster.c

```diff
@@ -155,6 +155,9 @@ bool am_mirror = false;
 /* GPDB specific flag to handle deadlocks during parallel segment start. */
 volatile bool *pm_launch_walreceiver = NULL;
 
+/* This is set by a test to simulate long RESET state. */
+static int	delay_bg_writer_termination_sec = 0;
```

Comment:
> Maybe postmaster_no_sigkill should be defined here as well, and put behind a #ifdef FAULT_INJECTOR?

If it was just one single piece of code to wrap, I would agree.
But in this patch there are several places changed, and wrapping every of them significantly reduces code readability .
It will lead to code like below:
```
#ifdef FAULT_INJECTOR
		if (delay_bg_writer_termination_sec == 0)
#endif
		{
			ereport(DEBUG2,
					(errmsg_internal("sending %s to process %d",
									 (SendStop ? "SIGSTOP" : "SIGQUIT"),
									 (int) BgWriterPID)));
			signal_child(BgWriterPID, (SendStop ? SIGSTOP : SIGQUIT));
		}
#ifdef FAULT_INJECTOR
		else
		{
			/*
			 * Code below is only for test purposes and should not be invoked
			 * under normal conditions. It will delay the termination of the
			 * background writer process.
			 *
			 * In order to delay the termination of the background writer
			 * process - send SIGSTOP to the bg writer process and start a
			 * timer that will wake up and then immediately terminate it.
			 */
			static TimeoutId id = 0;

			ereport(DEBUG2,
					(errmsg_internal("sending SIGSTOP to process %d and starting timer",
									 (int) BgWriterPID)));
			signal_child(BgWriterPID, SIGSTOP);

			if (id == 0)
				id = RegisterTimeout(USER_TIMEOUT, TerminateBgWriterHandler);

			enable_timeout_after(id, delay_bg_writer_termination_sec * 1000);
		}
#endif
```
I'd like to avoid such style if possible. Readability is important.